netket/operator/_continuous_operator.py

```diff
@@ -28,16 +28,18 @@ class ContinuousOperator(AbstractOperator):
     `ContinuousOperator` and implement its interface.
     """
 
-    def __init__(self, hilbert: AbstractHilbert, dtype: DType = float):
+    def __init__(self, hilbert: AbstractHilbert, dtype: Optional[DType] = None):
         r"""
         Constructs the continuous operator acting on the given hilbert space and
         with a certain data type.
 
         Args:
             hilbert: The underlying Hilbert space on which the operator is defined
-            dtype: Data type of the matrix elements. Defaults to `np.float64`
+            dtype: Data type of the operator, which is used to infer the dtype of
+                expectation values. Defaults to `float`
         """
-
+        if dtype is None:
+            dtype = float
```

Comment:
Why don't you leave the default argument in the init signature?

We could also think about leaving here a `None` default type that is the weakest and gets upgraded to anything else whenever needed.

Reply:
If I understand correctly, we should not write default arguments in the init signature of a base class, because if a subclass has `__init__(..., dtype=None)`, `None` will overwrite `float` here. Rather, we should handle `None` in the body of `__init__`.

(Maybe Gabriel didn't realize it when he wrote this. Dataclasses help solve it but they have other problems...)

Reply:
I think it looks strange to leave `op.dtype == None`, because a np/jnp array will never have `a.dtype == None`.

As for float vs float32, now I agree that float32 looks better, because it's also impossible to have `a.dtype == float`. Now I let `jax.dtypes.canonicalize_dtype(float)` to decide whether to use float32 or float64 here.

Reply:
great idea, thanks